Libraries/Core/Devtools/setupDevtools.js

```diff
@@ -15,96 +15,17 @@ var NativeModules = require('NativeModules');
 var Platform = require('Platform');
 
 function setupDevtools() {
-  var messageListeners = [];
-  var closeListeners = [];
-  var hostname = 'localhost';
-  if (Platform.OS === 'android' && NativeModules.AndroidConstants) {
-    hostname = NativeModules.AndroidConstants.ServerHost.split(':')[0];
-  }
-  var port = window.__REACT_DEVTOOLS_PORT__ || 8097;
-  var ws = new window.WebSocket('ws://' + hostname + ':' +  port + '/devtools');
-  // this is accessed by the eval'd backend code
-  var FOR_BACKEND = { // eslint-disable-line no-unused-vars
+  var {connectToDevTools} = require('react-devtools-core');
+  connectToDevTools({
+    // Special case: Genymotion is running on a different host.
+    host: (Platform.OS === 'android' && NativeModules.AndroidConstants) ?
+      NativeModules.AndroidConstants.ServerHost.split(':')[0] :
+      'localhost',
+    // Read the optional global variable for backward compatibility.
+    // It was added in https://github.com/facebook/react-native/commit/bf2b435322e89d0aeee8792b1c6e04656c2719a0.
+    port: window.__REACT_DEVTOOLS_PORT__,
```

Comment:
It's [here now](https://github.com/facebook/react-devtools/blob/309c926365483f62ce2420e0baaa7eef26c76da0/packages/react-devtools-core/src/backend.js#L47).